### ISSUE TITLE
Module slow to load

### ISSUE THREAD

**Issue by RBrearton** [nexpy/nexusformat#153]
This module is very slow to load. Running, e.g.,

```python
from time import process_time

start = process_time()
from nexusformat.nexus.tree import nxload
print(process_time() - start)
```

will tell you the bad news on your own machine. I have two questions on this topic:

1) What is the purpose of the line:
```python
from .completer import nxcompleter
```
in nexusformat.nexus.__init__.py? Is this so central to the module that users are forced to import it? Is it truly **_always_** required? I ask because it seems to be working some autocomplete magic that I'm unfamiliar with, but all the tests pass with that line of code commented out. This means that either (a) the tests are incomplete or (b) this is optional and should not be run automatically.

On my machine, this import (namely the IPython imports it implies) takes up more than half of the import time.

2) Are there any plans to refactor this module, or otherwise improve load times? It seems strange to me that I have to run a 7000 line file to import e.g. nxload – this lack of modularity must contribute to slow load times. There are other obvious issues with having the entire functionality of the package stored in one .py file, but I suppose such a discussion would be better saved for another issue.

I should finish by emphasising that I would be very happy to help with refactoring the module, if this is not an ongoing effort. It's a great package; I use it regularly, and plan to continue to do so!

**Comment by rayosborn** [comment]
@RBrearton, I have merged #154 in response to this before making some other changes and then checking your other PR #152. Let me know if it works. If not, feel free to submit another PR.

**Comment by rayosborn** [comment]
Fixed in #154 and #155.
